### PR TITLE
NEW: Dice Metric

### PR DESCRIPTION
**Pull request by ruqayya**
- Add a function is added for computing Sørensen–Dice coefficient.

### CODE REVIEW

File: tiatoolbox/utils/metrics.py [hunk 1]
```diff
@@ -66,3 +66,27 @@ def f1_detection(true, pred, radius):
...
+
+    Args:
+            gt_mask (:class:`numpy.ndarray`):
+                    A binary ground truth mask
+            pred_mask (:class:`numpy.ndarray`):
+                    A binary predicted mask
+
+    Returns:
+            :class:`float`:
+                    A dice overlap
```
Comment (John-P) [inline tiatoolbox/utils/metrics.py:82]:
```suggestion
                    A dice overlap

```

File: tiatoolbox/utils/metrics.py [hunk 2]
```diff
@@ -66,3 +66,27 @@ def f1_detection(true, pred, radius):
     fp = len(unpaired_pred)
     fn = len(unpaired_true)
     return tp / (tp + 0.5 * fp + 0.5 * fn)
+
+
+def dice(gt_mask, pred_mask):
+    """This function computes dice coefficient between two masks.
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tiatoolbox/utils/metrics.py:72]: @John-P  please can you provide a permalink to a line in docstring where you have added maths equations? That should help @ruqayya to add this.

**Review by shaneahmed — COMMENTED** [review]
Thanks @ruqayya Looks good to me, if you can address the comments we should be able to merge.

File: tiatoolbox/utils/metrics.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (John-P) [reply tiatoolbox/utils/metrics.py:72]: https://github.com/TissueImageAnalytics/tiatoolbox/blob/8e316449fc899e1bda6f20f8b59bb984663c9d6c/tiatoolbox/utils/transforms.py#L159

[the remark this answers is not in the document]
Comment (ruqayya) [reply tiatoolbox/utils/metrics.py]: Added here https://github.com/TissueImageAnalytics/tiatoolbox/blob/dev-dice_metric/tiatoolbox/utils/metrics.py#:~:text=def%20dice(,Y%7C%20/%20%7CX%7C%20%2B%20%7CY%7C

[the remark this answers is not in the document]
Comment (ruqayya) [reply tiatoolbox/utils/metrics.py]: Added here https://github.com/TissueImageAnalytics/tiatoolbox/blob/dev-dice_metric/tiatoolbox/utils/metrics.py#L72:~:text=def%20dice(,Y%7C%20/%20%7CX%7C%20%2B%20%7CY%7C

File: tests/test_metrics.py [hunk 3]
```diff
@@ -24,3 +25,15 @@ def test_f1_detection():
     set_b = np.array([[0.1, 0.1], [1.1, 1.1], [2.1, 2.1], [3.1, 3.1], [4.2, 4.2]])
     score = f1_detection(set_a, set_b, 0.2)
     assert score - 1.0 < 1.0e-6
+
+
+def test_dice():
+    """Test to calculate DICE."""
+    gt_mask = np.random.randint(2, size=(256, 256))
+    pred_mask = np.random.randint(2, size=(256, 256))
+    _ = dice(gt_mask, pred_mask)
```
[the remark this answers is not in the document]
Comment (ruqayya) [reply tests/test_metrics.py]: I have added it for checking range between 0 and 1. https://github.com/TissueImageAnalytics/tiatoolbox/blob/dev-dice_metric/tests/test_metrics.py#:~:text=assert%20dice_val%20%3E%3D,dice_val%20%3C%3D%201.0

**Review by shaneahmed — APPROVED** [review]

**Pull request merged**